### PR TITLE
chore(release): adapt release-it config to remove/fix deprecated options as from version 10

### PR DESCRIPTION
**Pull request by christophercr**
ISSUES CLOSED: #1236 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[X] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #1236 

## What is the new behavior?
The release-it config is migrated as indicated in the warnings so no more deprecation warnings are displayed when running the `npm run release` command.

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

## Other information
Took the chance to re-generate all the `package-lock.json` files to upgrade all the dependencies to the latest version possible.

### CODE REVIEW

File: .release-it.json [hunk 1]
```diff
@@ -6,13 +6,12 @@
...
-		"afterBump": "npm run check:nightly:stark-versions -- ${version} && npm run generate:changelog",
 		"afterRelease": false,
-		"beforeStage": false,
-		"beforeStart": "npm run check:starter:stark-versions",
-		"changelog": "npm run generate:changelog-recent"
+		"beforeStage": "npm run check:nightly:stark-versions -- ${version} && npm run generate:changelog",
+		"beforeStart": "npm run check:starter:stark-versions"
 	},
 	"git": {
+		"changelog": "npm run generate:changelog-recent",
```
Comment (SuperITMan) [inline .release-it.json]: Why did you move the "changelog" script in "git" instead of let it in "scripts" ?

Comment (christophercr) [reply .release-it.json]: Because of: https://github.com/release-it/release-it/commit/b2f0fc2a0459a5f670fdb260d57432033326c588

Comment (SuperITMan) [reply .release-it.json]: Oops. I took a look on the releases but I didn't see it ^^
